src/WCSimWCTrigger.cc

```diff
@@ -406,6 +409,210 @@ void WCSimWCTriggerBase::FillDigitsCollection(WCSimWCDigitsCollection* WCDCPMT,
+	    continue;
+
+	  //int parentID    = (*WCDCPMT)[i]->GetParentID(ip);
+
+	  //get the composition information for the triggered digit
+	  //WCDCPMT stores this information in pairs of (digit id, photon id)
+	  //need to loop to ensure we get all the photons associated with the current digit (digit ip)
+	  std::vector< std::pair<int,int> > digitized_composition = (*WCDCPMT)[i]->GetDigiCompositionInfo();
+	  std::vector<int> triggered_composition;
+	  for(std::vector< std::pair<int,int> >::iterator it = digitized_composition.begin(); it != digitized_composition.end(); ++it) {
+	    if((*it).first == ip) {
+	      triggered_composition.push_back((*it).second);
+	    }
+	    else if ((*it).first > ip)
+	      break;
+	  }//loop over digitized_composition
+
+#ifdef WCSIMWCTRIGGER_VERBOSE
+	  G4cout << "Saving digit on PMT " << tube
+		 << " time " << digihittime
+		 << " pe "   << peSmeared
+		 << " digicomp";
+	  for(unsigned int iv = 0; iv < triggered_composition.size(); iv++)
+	    G4cout << " " << triggered_composition[iv];
+	  G4cout << G4endl;
+#endif
+	  assert(triggered_composition.size());
+
+	  //add hit
+	  if ( DigiHitMap[tube] == 0) {
+	    //this PMT has no digits saved yet; create a new WCSimWCDigi
+	    WCSimWCDigiTrigger* Digi = new WCSimWCDigiTrigger();
+	    Digi->SetTubeID(tube);
+	    //Digi->AddParentID(parentID);
+	    Digi->AddGate  (itrigger);
+	    Digi->SetTime  (itrigger,digihittime);
+	    Digi->SetPe    (itrigger,peSmeared);
+	    Digi->AddPe    ();
+	    Digi->AddDigiCompositionInfo(itrigger,triggered_composition);
+	    DigiHitMap[tube] = DigitsCollection->insert(Digi);
+	  }
+	  else {
+	    //this PMT has digits saved already; add information to the WCSimWCDigi
+	    //(*DigitsCollection)[DigiHitMap[tube]-1]->AddParentID(parentID);
+	    (*DigitsCollection)[DigiHitMap[tube]-1]->AddGate(itrigger);
+	    (*DigitsCollection)[DigiHitMap[tube]-1]->SetTime(itrigger, digihittime);
+	    (*DigitsCollection)[DigiHitMap[tube]-1]->SetPe  (itrigger, peSmeared);
+	    (*DigitsCollection)[DigiHitMap[tube]-1]->AddPe  ();
+	    (*DigitsCollection)[DigiHitMap[tube]-1]->AddDigiCompositionInfo(itrigger,triggered_composition);
+	  }
+	  if(remove_hits)
+	    (*WCDCPMT)[i]->RemoveDigitizedGate(ip);
+	}//digits within trigger window
+      }//loop over Digits
+    }//loop over PMTs
+  }//loop over Triggers
+  G4cout << "WCSimWCTriggerBase::FillDigitsCollection. Number of entries in output digit collection: " << DigitsCollection->entries() << G4endl;
+
+}
+
+
+void WCSimWCTriggerBase::AlgNoTrigger(WCSimWCDigitsCollection* WCDCPMT, bool remove_hits, bool test) {
+
+  if( output_txt ){
+    ofstream myfile_hits;
+    int nchar = (ceil(log10(event_number))+1);
+    char * num =  (char*)malloc(sizeof(char)*nchar);
+    sprintf(num, "%d", event_number);
+    std::string event_file_base = "all_hits_";
+    std::string event_file_suffix = ".txt";
+    std::string event_file = event_file_base + num + event_file_suffix;
+    event_number++;
```

Comment:
event_number may be useful for other things. Maybe move the iteration to the Digitize() method?